brave/brave-ios
# Fix #6005: Wallet solana sign transactions

## Summary of Changes
Sign Solana Transaction and All Transactions requested from Solana Dapps.


This pull request fixes #6005

## Submitter Checklist:


## Test Plan:
1. Go to https://pwgoom.csb.app/
2. connect the dapp with brave wallet
3. click Sign Transaction 
4. observe the wallet notification will show up
5. click to open wallet (unlock wallet if it's locked)
6. observe a sign transaction panel will show up
7. check 
   a. network 
   b. dapp's origin 
   c. account 
   d. check warning message has been shown e. check CTA are cancel (cancel to sign transaction) and continue(continue to pass this warning message)
9. click cancel
10. observe the test dapp log displaying user has rejected this request
11. repeat step 3 to 6
12. click continue 
13. observe the transaction details shows up and new CTA sign shows up 
14. click sign
15. observe request panel dismissed and test dapp prints out `signTransaction` object with no error

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Repeat the above steps but for `sign all transactions`

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Test multiple requests:
1. Go to https://pwgoom.csb.app/
2. connect the dapp with brave wallet
3. click sign transaction
4. click notification and unlock wallet if its locked
5. observe a sign transaction request panel will show up 
6. dimiss it by swipe it down
7. click sign transaction again
8. observe a sign transaction request panel show up again but this time with `1 of x` Next on the top right (x will be the times you creates sign transaction request, in this case would be 2)
9. click next to see if it loops though the queue correctly 
10. observe that if the current request is not the very first one `Continue` button will be disabled
11. click Continue to pass the warning message
12. click next button on top right to see if it looks though the queue correcly
13. observe that if the current request is not the very first one `Sign` button will be disabled. 

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Repeat the above steps but for `sign all transactions`


## Screenshots:


![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-10-26 at 13 25 29](https://user-images.githubusercontent.com/1187676/198094778-00a2937c-53f2-4ab7-9778-a985f639af55.png) | ![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-10-26 at 13 25 31](https://user-images.githubusercontent.com/1187676/198094781-163fa416-0f40-41c9-8a23-25707484b8a2.png)
-- | --

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
